src/JsonApiDotNetCore.MongoDb/Internal/MongoDbModel.cs

```diff
@@ -0,0 +1,32 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using JsonApiDotNetCore.Configuration;
+using Microsoft.EntityFrameworkCore.Infrastructure;
+using Microsoft.EntityFrameworkCore.Metadata;
+
+namespace JsonApiDotNetCore.MongoDb.Internal
```

Comment:
Top-level Internal namespaces are an anti-pattern, similar to Shared, Common, Utilities etc. because they tend to be filled with unrelated types (which defeats the purpose of having namespaces). Nested Internal namespaces are reserved for [pubternal](https://amp.reddit.com/r/dotnet/comments/jwzgl9/word_of_the_day_pubternal/) types, which these are not. Instead I recommend to put these, along with MongoDbRepository in JsonApiDotNetCore.MongoDb.Repositories.

Reply:
> Instead I recommend to put these, along with MongoDbRepository in JsonApiDotNetCore.MongoDb.Repositories

Did you intentionally place them in the project root instead?

You've placed `MongoDbQueryExpressionValidator` in `JsonApiDotNetCore.MongoDb.Queries.Expressions`, which is the namespace where immutable expression objects live. This is not one of those, instead it is validation logic that is closely related to the repository and not a generic building block, so I would put it there too.

Reply:
Also, please move GettingStarted and Example projects inside Examples subfolder, similar to the main repo.

Reply:
In my last commit I did put all these classes in `JsonApiDotNetCore.MongoDb.Repositories`, sorry about that!